### PR TITLE
Add Matrix channel and Libera.Chat channel as unaffiliated spaces

### PR DESCRIPTION
**Pull request by jgarte**
What do you think of adding the purescript matrix channel as an addition/alternative to slack.

#purescript:matrix.org

### CODE REVIEW

**Comment by MonoidMusician** [comment]
My reservations about adding this to the README are that only a handful of people are active there, only a few people in the room altogether whom I recognize as active within the PS community, and I'm not sure about moderation in that channel (it looks like there's only one default admin who created it). Of course advertising it could help with the first part, but I'd like to solve the second part too.

**Comment by thomashoneyman** [comment]
I think we either should only include official channels in the README (ie. places where recognized moderators have agreed to enforce the code of conduct), or should create a separate section for unofficial channels (where the code of conduct is not necessarily enforced). The only places where we promise to enforce the code of conduct are Discourse, Discord, and PureScript properties on GitHub.

**Comment by jgarte** [comment]
> My reservations about adding this to the README are that only a handful of people are active there, only a few people in the room altogether whom I recognize as active within the PS community, and I'm not sure about moderation in that channel (it looks like there's only one default admin who created it). Of course advertising it could help with the first part, but I'd like to solve the second part too.

Yes, maybe we can do the work to find out more about both of your points/concerns and see what the community wants today. I would like a bigger community with more open platforms for live chat choices besides just slack. I think only good can come from this. 

Common Lisp, for example, which has a relatively small community, has numerous channels on matrix, [irc](https://irclog.tymoon.eu/), and [other platforms](https://shirakumo.github.io/lichat) moderated by various people.

I think that zig is a good example of [this approach](https://github.com/ziglang/zig/wiki/Community).

On zig's website they state `Join a Community` and then promote and encourage various choices for learning with various communities of zig enthusiasts.

IIRC, purescript arose out of this ethos of giving the users more choice in what they can do with a language so why not also reflect that spirit by encouraging more open communication platforms such as matrix, irc, xmpp, etc...? I think that this has the potential to grow the community even larger.

**Comment by thomashoneyman** [comment]
I'm not against listing various PureScript communities to choose from, but I want to be clear about which ones the PureScript team moderates (ie. where we promise to uphold our [Code of Conduct](https://github.com/purescript/governance/blob/master/CODE_OF_CONDUCT.md)) and which ones we don't moderate. If we are going to add more options to the README, then I think they should be kept in a separate list from the official communication channels.

**Comment by jgarte** [comment]
> I'm not against listing various PureScript communities to choose from, but I want to be clear about which ones the PureScript team moderates (ie. where we promise to uphold our [Code of Conduct](https://github.com/purescript/governance/blob/master/CODE_OF_CONDUCT.md)) and which ones we don't moderate. If we are going to add more options to the README, then I think they should be kept in a separate list from the official communication channels.

That's a good point. I asked in https://matrix.to/#/#purescript:matrix.org if they have a code of conduct. Would be good to mention it.

**Comment by MonoidMusician** [comment]
I'm not clear on what you mean by “open platform” – in what ways do you consider Matrix or Slack open platforms? Also, most discussions are on Discord these days not Slack.

**Comment by jgarte** [comment]
> I'm not clear on what you mean by “open platform” – in what ways do you consider Matrix or Slack open platforms? Also, most discussions are on Discord these days not Slack.

By open I was referring to open source/free software since slack is proprietary software.
Sorry if that could have been interpreted another way.

**Comment by MonoidMusician** [comment]
Oh okay, that makes sense! I just misinterpreted it by assuming you were including Slack in that.

I did suggest creating a Zulip for PureScript back when we were discussing alternatives, but more of the community was in favor of Discord at the time (due to familiarity). I am on a private Matrix now too, it's still catching up to Zulip in terms of features.

**Comment by thomashoneyman** [comment]
We migrated from Slack to Discord last year for several reasons, which you can read about here:

https://discourse.purescript.org/t/migrating-to-discord

I remember evaluating Matrix at the time — I would rather use an open platform than Discord — but for various reasons we decided against it. I think at the time no matrix client was as polished (for example, no support for video / audio chat, scheduled events, etc) or had sufficient moderation features.

I’d eventually like to switch to an open platform.

**Comment by JordanMartinez** [comment]
> I’d eventually like to switch to an open platform.

I agree with this. Discord was a "best overall tradeoff" at the time, but still not the ideal solution we wanted. I expect that to change in the future.

**Comment by JordanMartinez** [comment]
> If we are going to add more options to the README, then I think they should be kept in a separate list from the official communication channels.

I also agree with this.

**Comment by jgarte** [comment]
> > If we are going to add more options to the README, then I think they should be kept in a separate list from the official communication channels.
> 
> I also agree with this.

Should I put the matrix channel in a sub/separate list and update the PR?

What should the subsection/list be called?

**Comment by JordanMartinez** [comment]
> 
> Should I put the matrix channel in a sub/separate list and update the PR?
> 
> What should the subsection/list be called?

How about we split the `Help!` section into two areas, so that it reads like this:

> ## Help!
>
> ### Official Platforms
>
>The following platforms are governed by the [PureScript Community Code of Conduct](https://github.com/purescript/governance/blob/master/CODE_OF_CONDUCT.md). The majority of PureScript users use these platforms discuss and collaborate on PureScript-related topics:
> - [PureScript Discord](https://purescript.org/chat)
> - [PureScript Discourse](https://discourse.purescript.org/)
> 
> ### Unofficial Platforms
> 
> Some PureScript users also collaborate on the below platforms. These are "unofficial" because they do not fall under the code of conduct linked above. They may have no code of conduct or one very different than the one linked above.
> - [PureScript Matrix](https://matrix.to/#/#purescript:matrix.org)
> - [PureScript on StackOverflow](http://stackoverflow.com/questions/tagged/purescript)

In the above rendering, I'm doing the following:
- not describing the Matrix community as small. If it grows over time, we won't have to update this file to remove 'small' or something.
- moving SO into 'unofficial platforms' as that's likely more correct
- not calling the above section "links" or "resources" as some might think they can add _their_ unofficial "monad tutorial" there and otherwise spam the section.
- clarifying that some places may not even have a code of conduct.

**Comment by rhendric** [comment]
The CoC states that it applies in ‘all community spaces’, and then goes on to enumerate the places it considers to be such (Discord, Discourse, and the various official GitHub orgs). We should probably re-use that language to minimize confusion: the ‘official platforms’ are ‘community spaces’, and the ‘unofficial platforms’ are... uh, something else? ‘Unaffiliated spaces’?

**Comment by JordanMartinez** [comment]
> The CoC states that it applies in ‘all community spaces’, and then goes on to enumerate the places it considers to be such (Discord, Discourse, and the various official GitHub orgs). We should probably re-use that language to minimize confusion: the ‘official platforms’ are ‘community spaces’, and the ‘unofficial platforms’ are... uh, something else? ‘Unaffiliated spaces’?

Sounds fine to me!

**Comment by toastal** [comment]
Would adding the `#purescript` room an Libera.Chat also be a worthwhile addition?

**Comment by jgarte** [comment]
> Would adding the `#purescript` room an Libera.Chat also be a worthwhile addition?

That would be great! I'd join it and would be willing to publicly log the channel here:

http://litterbox.whereis.みんな/

**Comment by JordanMartinez** [comment]
I think adding the Libera.Chat is also fine.

**Comment by JordanMartinez** [comment]
Due to the lack of activity on this, I'm going to push the last few commits to get this over the line.

**Comment by JordanMartinez** [comment]
Can this get an approval?

File: README.md
Comment (rhendric) [inline README.md:30]: ‘on the below spaces’ sounds wrong to me; ‘in ... spaces’ or ‘on ... platforms’?

**Review by rhendric — APPROVED** [review]
Looks good, only one nitpick.

I'm probably going to tweak the prerelease deployment criteria per [this comment](https://github.com/purescript/purescript/issues/4323#issuecomment-1126744100); maybe hold off on merging this until then?

File: README.md
Comment (JordanMartinez) [reply README.md:30]: Yeah... I was wondering about this myself. I suppose we could change the CoC to use the term 'platform' instead, but...

Comment (JordanMartinez) [reply README.md:30]: I'll push the 'in spaces' as the fix.

And yeah, let's hold off on this until that comment is addressed.

**Pull request merged**
